workflow/validate/validate.go

```diff
@@ -548,7 +548,8 @@ func resolveAllVariables(scope map[string]interface{}, tmplStr string) error {
 			return nil
 		}
 		_, ok := scope[tag]
-		if !ok {
+		_, isGlobal := globalParams[tag]
+		if !ok && !isGlobal {
```

Comment:
tests?

Reply:
Sorry. Added a test case to `workflow/validate/validate_test.go` and verified that it failed without the changes in this PR.